python/lsst/daf/butler/core/location.py

```diff
@@ -395,7 +398,9 @@ def relativeToPathRoot(self):
         Effectively, this is the path property with posix separator stipped
         from the left hand side of the path.
         """
-        return self.path.lstrip("/")
+        p = PurePosixPath(os2posix(self.path))
```

Comment:
`self.path` returns OS path for `file` and scheme-less but POSIX path for everything else. You need to deal with that.